R/nm-file.R

```diff
@@ -87,20 +88,31 @@ nm_par_tab <- function(.mod) {
 
 #' @describeIn nm_file Reads the input data file from a `bbi_nonmem_model` or
 #'   `bbi_nonmem_summary` object
+#' @param filter Logical (`T`/`F`). If `TRUE`, filter data based on `IGNORE LIST`
+#'  or `ACCEPT LIST` options defined in the `$DATA` record.
 #' @importFrom data.table fread
 #' @importFrom tibble as_tibble
 #' @export
-nm_data <- function(.mod) {
+nm_data <- function(.mod, filter = FALSE) {
   check_model_object(.mod, c(NM_MOD_CLASS, NM_SUM_CLASS, NMSIM_MOD_CLASS))
   .path <- get_data_path(.mod)
   verbose_msg(glue("Reading data file: {basename(.path)}"))
   .d <- fread(.path, na.strings = ".", verbose = FALSE)
   .d <- remove_dup_cols(.d)
   .d <- as_tibble(.d)
+
   names(.d) <- toupper(names(.d))
+
+  if(isTRUE(filter)){
+    .d <- filter_nm_data(.mod, data = .d)
+    recs_dropped <- attributes(.d)$n_records_dropped
+    verbose_msg(paste("Number of records dropped:", cli::col_blue(recs_dropped)))
```

Comment:
Probably saying the number retained will be more helpful here, if anything is printed at all.